tests/tracing/test_otel_logging.py

```diff
@@ -336,3 +337,295 @@ def test_empty_resource_spans_returns_400(mlflow_server: str):
+            span.trace_id = bytes.fromhex(trace_id_hex)
+            span.span_id = bytes.fromhex(f"{trace_id_num * 100 + span_num:08x}" + "0" * 8)
+            span.name = f"mixed-span-{trace_name}-{span_num}"
+            span.start_time_unix_nano = 1000000000 + span_num * 1000
+            span.end_time_unix_nano = 2000000000 + span_num * 1000
+
+            scope = InstrumentationScope()
+            scope.name = "mixed-test-scope"
+
+            scope_spans = ScopeSpans()
+            scope_spans.scope.CopyFrom(scope)
+            scope_spans.spans.append(span)
+
+            resource = Resource()
+            resource_spans = ResourceSpans()
+            resource_spans.resource.CopyFrom(resource)
+            resource_spans.scope_spans.append(scope_spans)
+
+            request.resource_spans.append(resource_spans)
+
+    response = requests.post(
+        f"{mlflow_server}/v1/traces",
+        data=request.SerializeToString(),
+        headers={
+            "Content-Type": "application/x-protobuf",
+            MLFLOW_EXPERIMENT_ID_HEADER: experiment_id,
+        },
+        timeout=10,
+    )
+
+    assert response.status_code == 200
+
+    traces = mlflow.search_traces(
+        experiment_ids=[experiment_id], include_spans=True, return_type="list"
+    )
+
+    assert len(traces) == 3
+
+    # Verify span counts per trace
+    # Note: search_traces may return traces in any order, so we need to check by span count
+    span_counts = sorted([len(trace.data.spans) for trace in traces[:3]])
+    expected_counts = sorted([2, 1, 2])  # trace A: 2, trace B: 1, trace C: 2
+
+    assert span_counts == expected_counts
+
+
+def test_error_logging_spans(mlflow_server: str, monkeypatch):
+    mlflow.set_tracking_uri(mlflow_server)
+    experiment = mlflow.set_experiment("otel-error-test")
+    experiment_id = experiment.experiment_id
+
+    resource = OTelSDKResource.create({"service.name": "test-batch-service"})
+    tracer_provider = TracerProvider(resource=resource)
+
+    exporter = OTLPSpanExporter(
+        endpoint=f"{mlflow_server}/v1/traces",
+        headers={MLFLOW_EXPERIMENT_ID_HEADER: experiment_id},
+        timeout=10,
+    )
+
+    # Use BatchSpanProcessor to batch spans from multiple traces
+    span_processor = BatchSpanProcessor(exporter)
+    tracer_provider.add_span_processor(span_processor)
+
+    # Reset the global tracer provider
+    otel_trace._TRACER_PROVIDER_SET_ONCE = Once()
+    otel_trace._TRACER_PROVIDER = None
+    otel_trace.set_tracer_provider(tracer_provider)
+
+    tracer = otel_trace.get_tracer(__name__)
+
+    count = 0
+
+    original_log_spans = SqlAlchemyStore.log_spans
+
+    def mock_log_spans(self, *args, **kwargs):
+        nonlocal count
+        if count == 0:
+            count += 1
+            raise Exception("test_error")
+        else:
+            return original_log_spans(self, *args, **kwargs)
+
+    monkeypatch.setattr(SqlAlchemyStore, "log_spans", mock_log_spans)
```

Comment:
can we use `mock.patch` instead? `monkeypatch.setattr` leads to patches being active longer than needed